R/Rules-class.R

```diff
@@ -367,50 +367,56 @@ IncrementsRelative <- function(intervals,
                         increments=increments)
 }
 
-## --------------------------------------------------
-## Increments control based on number of dose levels
-## --------------------------------------------------
+# nolint end
 
-##' Increments control based on number of dose levels
-##'
-##' @slot maxLevels scalar positive integer for the number of maximum
-##' dose levels to increment for the next dose. It defaults to 1,
-##' which means that no dose skipping is allowed - the next dose
-##' can be maximum one level higher than the current dose.
-##'
-##' @example examples/Rules-class-IncrementsNumDoseLevels.R
-##' @export
-##' @keywords classes
-.IncrementsNumDoseLevels <-
-  setClass(Class="IncrementsNumDoseLevels",
-           representation(maxLevels="integer"),
-           prototype(maxLevels=1L),
-           contains="Increments",
-           validity=
-             function(object){
-               o <- Validate()
+# IncrementsNumDoseLevels-class ----
 
-               o$check(is.scalar(object@maxLevels) &&
-                         is.integer(object@maxLevels) &&
-                         object@maxLevels > 0,
-                       "maxLevels must be scalar positive integer")
+#' `IncrementsNumDoseLevels`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' @slot maxLevels (`count`)\cr corresponding to the number of maximum
+#'   dose levels to increment for the next dose. It defaults to 1,
+#' which means that no dose skipping is allowed - the next dose
+#' can be maximum one level higher than the current dose.
+#' @slot basisLevel (`string`)\cr corresponding to the dose level used to increment from.
+#' It can take two possible values 'lastGiven' or 'maxGiven'. If lastGiven (default)
```

Comment:
```suggestion
#' It can take two possible values `lastGiven` or `maxGiven`. If `lastGiven` (default)
```
also below - the reason is that 1) this formats the string as code, 2) the spell check is not run on this text (because it is recognized as code).